src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -1015,13 +1015,20 @@ private void createSymlinks(Iterable<SymlinkMetadata> symlinks) throws IOExcepti
               "Failed creating directory and parents for %s",
               symlink.path())
           .createDirectoryAndParents();
-      // If a directory output is being materialized as a symlink, we must first delete the
-      // preexisting empty directory.
-      if (symlink.path().exists(Symlinks.NOFOLLOW)
-          && symlink.path().isDirectory(Symlinks.NOFOLLOW)) {
+      // If a directory output is being materialized as a symlink, creating the symlink fails as we
+      // must first delete the preexisting empty directory. Since this is rare (and in the future
+      // BwoB may no longer eagerly create these directories), we don't delete the directory
```

Comment:
This refers to https://github.com/bazelbuild/bazel/pull/25976. I can update the comment depending on which PR is merged first.

Reply:
Approving since #25976 can't yet be merged.